.github/workflows/ci-tier2.yml

```diff
@@ -0,0 +1,179 @@
+# CI for Tier2 platforms
+# This should not block CI or Release 
+
+name: CI-Tier2
+
+permissions:
+  contents: read
+
+on:
+  pull_request:
+    branches: [master]
+  workflow_dispatch:
+    inputs:
+      verbose:
+        description: "Set --verbose to get verbose build output"
+        required: false
+        default: ''
+
+env:
+  VERBOSE: ${{ github.events.input.verbose }}
+
+jobs:
+  cancel_previous_runs:
+    name: Cancel Previous Runs
+    runs-on: ubuntu-latest
+    steps:
+      - uses: styfle/cancel-workflow-action@0.4.1
+        with:
+          access_token: ${{ secrets.GITHUB_TOKEN }}
+
+  build_checks:
+    name: ${{ matrix.task.name }} (${{ matrix.os }})
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-latest]
+        rust: [stable]
+        task:
+          - name: Doc tests
+            run: make run-all-doc-test
+        include:
+          - os: ubuntu-latest
+            rust: stable
+            sccache-path: /home/runner/.cache/sccache
+            target: x86_64-unknown-linux-musl
+            task:
+              name: Rustfmt
+              run: make check-fmt
+          - os: macos-latest
+            rust: stable
+            sccache-path: /Users/runner/Library/Caches/Mozilla.sccache
+            target: x86_64-apple-darwin
+            task:
+              name: Clippy
+              run: make check-clippy
+          - os: ubuntu-latest
+            sccache-path: /home/runner/.cache/sccache
+            target: x86_64-unknown-linux-musl
```

Comment:
This should all be able to be gotten rid of, since rustfmt, clippy, and ubuntu-latest are run on Tier 1 CI